### PR TITLE
Service Catalog REST API fixes

### PR DESCRIPTION
**Pull request by vithu30**
This PR adds the following fixes:

- Fixes the issue in scope validation of Service Catalog REST APIs
- Removes unused scopes
- Changes the request path from `/service-entries` to `/services` - Fixes https://github.com/wso2/product-apim/issues/9841
- Changes the default value of shrink to false - Fixes wso2/product-apim#9842
- Updates the `importServices` resource implementation
- Adds the `DELETE /services/{serviceId}` resource - Fixes https://github.com/wso2/product-apim/issues/9837
- Adds `serviceKey` parameter to `Service` swagger model - Fixes https://github.com/wso2/product-apim/issues/9838

### CODE REVIEW

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java [hunk 1]
```diff
@@ -81,26 +82,7 @@ public String addServiceEntry(ServiceEntry serviceEntry, int tenantID, String uu
             boolean initialAutoCommit = connection.getAutoCommit();
```
[the remark this answers is not in the document]
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/9712/commits/55b71daae541457eba476461fe710aac4b67b029

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java [hunk 2]
```diff
@@ -113,7 +95,70 @@ public String addServiceEntry(ServiceEntry serviceEntry, int tenantID, String uu
...
+            if (isNewConnection) {
+                APIMgtDBUtil.closeAllConnections(null, connection, null);
+            }
+        }
+    }
+
+    public void updateServices(List<ServiceEntry> services, int tenantId, String username)
+            throws APIManagementException {
+        try (Connection connection = APIMgtDBUtil.getConnection();
+        PreparedStatement ps = connection.prepareStatement(SQLConstants.ServiceCatalogConstants
```
[the remark this answers is not in the document]
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java]: Fixed in https://github.com/wso2/carbon-apimgt/commit/55b71daae541457eba476461fe710aac4b67b029

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java [hunk 3]
```diff
@@ -113,7 +95,70 @@ public String addServiceEntry(ServiceEntry serviceEntry, int tenantID, String uu
...
+                .DELETE_SERVICE_BY_SERVICE_KEY)) {
+            for (ServiceEntry service: services) {
+                ps.setString(1, service.getKey());
+                ps.setInt(2, tenantId);
+                ps.addBatch();
+            }
+            ps.executeBatch();
+            addServices(services, tenantId, username, connection);
+        } catch (SQLException e) {
```
[the remark this answers is not in the document]
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java]: Fixed in https://github.com/wso2/carbon-apimgt/commit/55b71daae541457eba476461fe710aac4b67b029

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java [hunk 4]
```diff
@@ -1030,6 +1031,36 @@ public static OAuthApplicationInfo registerOAuthApplication(OAuthAppRequest appR
...
+    /**
+     * This is static method to return URI Templates map of Service Catalog REST API.
+     * This content need to load only one time and keep it in memory as content will not change
+     * during runtime.
+     *
+     * @return URITemplate set associated with Service Catalog REST API
+     */
+    public static Set<URITemplate> getServiceCatalogAPIResourceMapping() {
+        API api = new API(new APIIdentifier(RestApiConstants.REST_API_PROVIDER,
+                RestApiConstants.REST_API_SERVICE_CATALOG_CONTEXT_FULL, "v1"));
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java:1043]: Shall we make this v0.

**Review by malinthaprasan — APPROVED** [review]

**Review by tharindu1st — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/h2.sql [hunk 5]
```diff
@@ -2089,6 +2089,8 @@ CREATE TABLE IF NOT EXISTS AM_SERVICE_CATALOG (
             UPDATED_BY VARCHAR(255),
             ENDPOINT_DEFINITION BLOB NOT NULL,
             METADATA BLOB NOT NULL,
-            PRIMARY KEY (UUID)
+            PRIMARY KEY (UUID),
```
Comment (tharindu1st) [inline features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/h2.sql:2092]: changes for other dbs missing

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java (hunk 4 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9712/commits/f68a31c1a2cfe0949a410eb5ed03f40bd1c5fbb7

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/h2.sql (hunk 5 above)
Comment (vithu30) [reply features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/h2.sql:2092]: Fixed in https://github.com/wso2/carbon-apimgt/pull/9712/commits/eae46b5f9ad76f1794a9642dff0b47904ebedf09

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**